client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -653,7 +662,16 @@ const mapDispatchToProps = {
 
 type ConnectedProps = ReturnType< typeof mapStateToProps > & typeof mapDispatchToProps;
 
-export default connect(
-	mapStateToProps,
-	mapDispatchToProps
+export default compose(
+	connect(
+		mapStateToProps,
+		mapDispatchToProps
+	)
```

Comment:
select( 'core' ) always returns null in this context so can't use getEntityRecord to get the templateIds. Anybody know why this is the case?  Or maybe there is a much better way of getting the template wrappers back around the post content so don't need to worry about this?

Reply:
Oh I think this is quite twisted!
Basically, there are two (well more, but two are relevant in this case) separate Redux stores going on, one for Calypso in the top frame, and one for Gutenberg inside its own wp-admin iframe.
They communicate between each other through a pub/sub layer, as you might have noticed by scrolling this big file.

The revision dialog runs in the Calypso frame so it doesn't have direct access to the Gutenberg store, which is the one you are quering with the `select( 'core' )`.

One way we could fix this, I guess, would be to send the template info across this messaging system.

Another, would be to have Calypso fetch the template posts, store them in its own store, and use them as needed to fix the revisions.
We could do this with the [`<QueryPosts>`](https://github.com/Automattic/wp-calypso/tree/3f393deb9b402e240f47a638ae44b97819116cbf/client/components/data/query-posts) data component, querying for the `wp_template` post type, and then selecting them with [`getPostsForQuery`](https://github.com/Automattic/wp-calypso/blob/3f393deb9b402e240f47a638ae44b97819116cbf/client/state/posts/selectors.js#L140) ([here more infos on the query object](https://github.com/Automattic/wp-calypso/blob/3f393deb9b402e240f47a638ae44b97819116cbf/client/lib/query-manager/post/index.js)).

Maybe something like this (untested code):
```jsx
// Add around the beginning of the CalypsoifyIframe render function
<QueryPosts siteId={ siteId } query={ { type: 'wp_template' } }  />

// In the mapStateToProps
return {
  ...,
  templateParts: getPostsForQuery( state, siteId, { type: 'wp_template' } )
};
```